### PR TITLE
DR-35: use liquibase to create tables

### PR DESCRIPTION
**Pull request by ahaessly**
liquibase is integrated in gradle.build and updates local postgres DB

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -28,7 +28,8 @@ buildscript {
 }
 
 plugins {
-  id "com.palantir.docker" version "0.20.1"
```
Comment (ahaessly) [inline build.gradle:31]: should we standardize on single quotes?

File: build.gradle [hunk 2]
```diff
@@ -142,5 +160,6 @@ docker {
 
 compileGeneratedJava.dependsOn generateApi
 classes.dependsOn generatedClasses
+compileJava.dependsOn update
 compileJava.dependsOn compileGeneratedJava
 ideaModule.dependsOn generateApi
```
Comment (ahaessly) [inline build.gradle]: do we want intellij specifics in our build?

File: build.gradle (hunk 1 above)
Comment (ddietterich) [reply build.gradle:31]: Quoting in gradle is similar to bash: double quotes allow interpolation; single quotes do not. For example, we might want something like this:
```
def palantir_version = '0.20.1'
plugins {
  id "com.palantir.docker" version "$palantir_version"
}
```

That said, I'm happy to use single quotes as the default unless there is a reason to use doubles.

File: DATABASE.md [hunk 3]
```diff
@@ -20,3 +20,9 @@ This version defaults to starting postgres server version 11. To change to versi
 ### Clear the database
 
 `pslq -f db/truncate-tables jadedb jade`
+
+
+### Gradle commands
+
+To run database updates (this is not done automatically), run `./gradlew update`
```
Comment (ddietterich) [inline DATABASE.md:26]: Nice!

File: build.gradle [hunk 4]
```diff
@@ -124,6 +129,19 @@ jar {
     from sourceSets.generated.output
 }
 
+liquibase {
+    activities {
+        main {
+            changeLogFile 'src/main/resources/db/changelog.xml'
+            url 'jdbc:postgresql://127.0.0.1:5432/jadedb'
+            username 'jade'
+            password 'jadepasswd'
```
Comment (ddietterich) [inline build.gradle]: Seems fine to have password in here now for dev. I don't think we will want a single password for all instances of the repository metadata in all repos. What do you think?

Maybe we make a deployment story about that in the backlog?

File: src/main/resources/db/changesets/20181120_dataset.xml [hunk 5]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.4.xsd">
+    <changeSet author="ahaessly" id="create_dataset_table">
+        <createTable tableName="dataset">
+            <column name="id" type="UUID">
```
Comment (ddietterich) [inline src/main/resources/db/changesets/20181120_dataset.xml:5]: Maybe you can give us a short tutorial on liquibase next week?

**Review by ddietterich — APPROVED** [review]

File: build.gradle (hunk 4 above)
Comment (ahaessly) [reply build.gradle]: yes, i think maybe we get these values from vault? i'll make a story

File: build.gradle (hunk 1 above)
Comment (thathert) [reply build.gradle:31]: Added a section in our SDLC notes about code standards: https://docs.google.com/document/d/1Dx4Zy_IcOzwb_2d4S7pP56MnLD_r6jpkAq9dsWpT0GA/edit#

Comment (thathert) [reply build.gradle]: Separately, this plugin needs to be enabled for dockerization to work. I'll update

File: DATABASE.md (hunk 3 above)
Comment (thathert) [reply DATABASE.md:26]: Is there a way to rollback an update or only update to a specific point?

**Review by thathert — APPROVED** [review]
Resolved the merge conflict and switched this over to the de-jade-ified names. Let me know if this looks good @RoriCremer and @ddietterich, then let's merge.

**Review by RoriCremer — APPROVED** [review]

**Pull request merged**